File: code/jvm/main/kotlin/com/isel/leic/ps/ionClassCode/domain/Assignment.kt
package com.isel.leic.ps.ionClassCode.domain

import java.sql.Timestamp

/**
 * Assigment Domain Interface
 */
data class Assignment(
    val id: Int,
    val classroomId: Int,
    val minElemsPerGroup: Int,
    val maxElemsPerGroup: Int,
    val maxNumberGroups: Int,
    val releaseDate: Timestamp,
    val description: String,
    val title: String,
)

File: code/js/src/domain/Assignment.ts
export interface AssignmentDomain {
    id: number,
    classroomId: number,
    minElemsPerGroup: number,
    maxElemsPerGroup: number,
    maxNumberGroups: number,
    releaseDate: Date,
    description: string,
    title: string,
}

export class AssignmentDomain {
    constructor(
        id: number,
        classroomId: number,
        minElemsPerGroup: number,
        maxElemsPerGroup: number,
        maxNumberGroups: number,
        releaseDate: Date,
        description: string,
        title: string,
    ) {
        this.id = id
        this.classroomId = classroomId
        this.minElemsPerGroup = minElemsPerGroup
        this.maxElemsPerGroup = maxElemsPerGroup
        this.maxNumberGroups = maxNumberGroups
        this.releaseDate = releaseDate
        this.description = description
        this.title = title
    }
}